weave

```diff
@@ -25,19 +25,19 @@ CIDR_REGEXP="$IP_REGEXP/[0-9]{1,2}"
 usage() {
     echo "Usage:"
     echo "weave setup"
-    echo "weave launch       [-password <password>] [-nickname <nickname>] [-iprange <cidr>] <peer> ..."
+    echo "weave launch       [-password <password>] [-nickname <nickname>] [-iprange <cidr>] [-defaultsubnet <cidr>] <peer> ..."
     echo "weave launch-dns   <cidr>"
     echo "weave launch-proxy [-H <docker_endpoint>] [--with-dns] [--with-ipam]"
     echo "weave connect      [--replace] [<peer> ...]"
     echo "weave forget       <peer> ..."
-    echo "weave run          [--with-dns] [<cidr> ...] <docker run args> ..."
-    echo "weave start        [<cidr> ...] <container_id>"
-    echo "weave attach       [<cidr> ...] <container_id>"
-    echo "weave detach       <cidr> [<cidr> ...] <container_id>"
+    echo "weave run          [--with-dns] [[--subnet] <cidr> ...] <docker run args> ..."
+    echo "weave start        [[--subnet] <cidr> ...] <container_id>"
+    echo "weave attach       [[--subnet] <cidr> ...] <container_id>"
+    echo "weave detach       [[--subnet] <cidr> ...] <container_id>"
     echo "weave dns-add      <ip_address> [<ip_address> ...] <container_id> [-h <fqdn>]"
     echo "weave dns-remove   <ip_address> [<ip_address> ...] <container_id> [-h <fqdn>]"
-    echo "weave expose       [<cidr> ...] [-h <fqdn>]"
-    echo "weave hide         [<cidr> ...]"
+    echo "weave expose       [[--subnet] <cidr> ...] [-h <fqdn>]"
+    echo "weave hide         [[--subnet] <cidr> ...]"
```

Comment:
this is not accurate; it implies that `--subnet` is specified *once*, in front of all `<cidr>`s.

Also, you don't mention the `default`.

I suggest introducing a new arg "type".